Update submodule tips

esteele opened the pull request:
Add an action to update the submodule tips that can be called by volto/restapi/api.

Closes #1214

esteele committed:
Add workflow to update all submodules

esteele committed:
Test build

esteele committed:
Use v1

esteele committed:
Fix commit

esteele committed:
Fix syntax

esteele committed:
Don’t fail if there’s nothing to commit

esteele committed:
Test run

esteele committed:
Update branches for merging

esteele committed:
Only on dispatch

stevepiercy commented on .github/workflows/update_submodule.yml:
```diff
@@ -0,0 +1,40 @@
+      # Pull latest Volto
+      - name: Pull latest Volto
+        run: |
+          cd submodules/volto
```
I am not sure whether `cd` resets to the root of `documentation` for each command. I guess we will see.

esteele replied:
It does.

stevepiercy commented on .github/workflows/update_submodule.yml:
```diff
@@ -0,0 +1,40 @@
+          git add submodules/plone.api submodules/plone.restapi submodules/volto
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          git diff --exit-code && git commit -m "Update submodules tip" && git push
```
I assume that this workflow only fires when there are diffs in the `docs/src/` directories of each submodule. For example, in Volto, Netlify has this configuration.

https://github.com/plone/volto/blob/5e6fab072500364dd239d44fa6bec76e627bb749/netlify.toml#L4-L5

esteele committed:
Confirm directory status

esteele committed:
Test with push

esteele committed:
test

esteele committed:
Fix branch

esteele committed:
Revert "Fix branch"

This reverts commit 00e75bef22256a6a23874607d43a22778269f380.

esteele committed:
Revert "test"

This reverts commit 722847a8394723b0664ebc26b6014f6abe482343.

esteele committed:
Revert "Test with push"

This reverts commit 705cf3790f2b027c3017d0d6cde374453d868303.

esteele committed:
Revert "Confirm directory status"

This reverts commit 2f7908691eedef449100db31e028168de5ace02a.

stevepiercy commented:
@esteele I just updated my local docs repo and its submodules, but when I do a `git status`:

```console
$ git status
On branch 6-dev
Your branch is up to date with 'origin/6-dev'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   submodules/plone.api (new commits)
        modified:   submodules/plone.restapi (new commits)
        modified:   submodules/volto (new commits)

no changes added to commit (use "git add" and/or "git commit -a")
```

I expected that this PR would also commit the submodules to `6-dev`. The [latest failure was from 2 days ago](https://github.com/plone/documentation/actions/runs/3255922707/jobs/5345744536), but there have been [more recent merges to Volto](https://github.com/plone/volto/actions). It looks like the submodule repos need to be fired on pushes to their `master` branches, for example see [volto](https://github.com/plone/volto/actions/runs/3255862511/workflow).

esteele commented:
Ok. I’ll look into it. On Oct 17, 2022, at 8:30 PM, Steve Piercy ***@***.***> wrote:﻿
@esteele I just updated my local docs repo and its submodules, but when I do a git status:
$ git status
On branch 6-dev
Your branch is up to date with 'origin/6-dev'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   submodules/plone.api (new commits)
        modified:   submodules/plone.restapi (new commits)
        modified:   submodules/volto (new commits)

no changes added to commit (use "git add" and/or "git commit -a")
I expected that this PR would also commit the submodules to 6-dev. The latest failure was from 2 days ago, but there have been more recent merges to Volto. It looks like the submodule repos need to be fired on pushes to their master branches, for example see volto.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>
